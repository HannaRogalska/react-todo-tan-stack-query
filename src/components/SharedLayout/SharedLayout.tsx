import { Suspense } from "react";
import { NavLink, Outlet } from "react-router-dom";

const SharedLayout = () => {
  return (
    <div className="flex flex-col min-h-screen">
      <header className="flex  justify-between items-center p-[30px]">
        <img
          className="w-[40px]"
          src="https://cdn-icons-png.freepik.com/256/13077/13077693.png?semt=ais_white_label"
        ></img>
        <ul className="flex justify-between items-center gap-2">
          <li>
            <NavLink to="/">Home</NavLink>
          </li>
          <li>
            <NavLink to="/form">Form</NavLink>
          </li>
          <li>
            <NavLink to="/todo">Todo</NavLink>
          </li>
        </ul>
      </header>
      <main className="flex-1">
        <Suspense fallback={<div>Loading...</div>}>
          <Outlet />
        </Suspense>
      </main>
      <footer className="p-2">
        <span>Â© 2025 Todo App</span>
      </footer>
    </div>
  );
};

export default SharedLayout;
