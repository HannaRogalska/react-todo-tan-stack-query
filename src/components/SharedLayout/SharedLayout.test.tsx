import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import SharedLayout from "./SharedLayout";

describe("Shared", () => {
    it("renders", () => {
        const { container } = render(<SharedLayout />)
        const header = screen.getByRole("banner");
        expect(header).toBeInTheDocument()
        const main = screen.getByRole("main");
        expect(main).toBeInTheDocument();
        const footer = screen.getByRole("contentinfo");
        expect(footer).toBeInTheDocument();
        expect(container).toMatchSnapshot()
    })
})

