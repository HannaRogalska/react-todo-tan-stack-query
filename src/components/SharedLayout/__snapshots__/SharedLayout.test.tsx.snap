// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shared > render comp 1`] = `
<div>
  <div>
    <header />
    <main />
    <footer />
  </div>
</div>
`;

exports[`Shared > render componen 1`] = `
<div>
  <div>
    <header />
    <main />
    <footer />
  </div>
</div>
`;

exports[`Shared > render components 1`] = `
<div>
  <div>
    <header />
    <main />
    <footer />
  </div>
</div>
`;

exports[`Shared > renders 1`] = `
<div>
  <div>
    <header />
    <main />
    <footer />
  </div>
</div>
`;
